### ISSUE TITLE
DataItem attributes mapping

### ISSUE THREAD

**Issue by afennani** [TrakHound/MTConnect.NET#7]
in the ReadXml in
MTConnect.MTConnectStreams.DataItemCollection<T>
the copy attributes is never cleared. attributes are carried from the siblings.
there should be copy.Attributes.RemoveAll(); in the loop

**Comment by PatrickRitchie** [comment]
I believe you are correct. The below should prevent the issue you described from happening should it not?

```c#
// Clear any previous Attributes
copy.Attributes.RemoveAll();

// Copy Attributes
foreach (XmlAttribute attribute in child.Attributes)
{
    var attr = doc.CreateAttribute(attribute.Name);
    attr.Value = attribute.Value;
    copy.Attributes.Append(attr);
}

// Copy Text
copy.InnerText = child.InnerText;

```

**Comment by PatrickRitchie** [comment]
This is addressed and resolved in v3.0.0
